### PR TITLE
docs(api): reviewed in-line docs of endpoints

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

**Review by kodumbeats — APPROVED** [review]

File: app/controllers/api/database.php [hunk 1]
```diff
@@ -149,9 +149,9 @@ function createAttribute($collectionId, $attribute, $response, $dbForInternal, $
...
-    ->param('read', null, new Permissions(), 'An array of strings with read permissions. By default no user is granted with any read permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.')
-    ->param('write', null, new Permissions(), 'An array of strings with write permissions. By default no user is granted with any write permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.')
+    ->param('permission', null, new WhiteList(['document', 'collection']), 'Permissions type model to use for reading documents in this collection. You can use collection-level permission set once on the collection using the `read` and `write` params, or you can set document-level permission where each document read and write params will decide who has access to read and write to each document individually. [learn more about permissions](https://appwrite.io/docs/permissions) and get a full list of available permissions.')
```
Comment (eldadfux) [inline app/controllers/api/database.php:152]: Why do we need the absolute URL here? Did we face any problems?

File: app/controllers/api/database.php [hunk 2]
```diff
@@ -206,8 +206,8 @@ function createAttribute($collectionId, $attribute, $response, $dbForInternal, $
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_COLLECTION_LIST)
     ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
-    ->param('limit', 25, new Range(0, 100), 'Results limit value. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
-    ->param('offset', 0, new Range(0, APP_LIMIT_COUNT), 'Results offset. The default value is 0. Use this param to manage pagination.', true)
+    ->param('limit', 25, new Range(0, 100), 'Maximum number of collection to return in response. Use this value to manage pagination. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
```
Comment (eldadfux) [inline app/controllers/api/database.php:209]: Might be a good idea to use the pagination related params description for linking to our new upcoming pagination section in the docs

File: app/controllers/api/database.php [hunk 3]
```diff
@@ -773,7 +773,7 @@ function createAttribute($collectionId, $attribute, $response, $dbForInternal, $
     ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_ATTRIBUTE_EMAIL)
-    ->param('collectionId', '', new UID(), 'Collection unique ID. You can create a new collection using the Database service [server integration](/docs/server/database#createCollection).')
+    ->param('collectionId', '', new UID(), 'Collection ID. You can create a new collection using the Database service [server integration](https:///docs/server/database#createCollection).')
```
Comment (eldadfux) [inline app/controllers/api/database.php:776]: Missing the domain name.
```suggestion
    ->param('collectionId', '', new UID(), 'Collection ID. You can create a new collection using the Database service [server integration](https://appwrite.io/docs/server/database#createCollection).')
```

File: app/controllers/api/functions.php [hunk 4]
```diff
@@ -262,13 +262,13 @@
     ->label('sdk.response.code', Response::STATUS_CODE_OK)
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_FUNCTION)
-    ->param('functionId', '', new UID(), 'Function unique ID.')
+    ->param('functionId', '', new UID(), 'Function ID.')
     ->param('name', '', new Text(128), 'Function name. Max length: 128 chars.')
-    ->param('execute', [], new ArrayList(new Text(64)), 'An array of strings with execution permissions. By default no user is granted with any execute permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.')
-    ->param('vars', [], new Assoc(), 'Key-value JSON object.', true)
+    ->param('execute', [], new ArrayList(new Text(64)), 'An array of strings with execution permissions. By default no user is granted with any execute permissions. [learn more about permissions](https://appwrite.io/docs/permissions) and get a full list of available permissions.')
+    ->param('vars', [], new Assoc(), 'Key-value JSON object that will be passed to the function as environment variables.', true)
```
Comment (eldadfux) [inline app/controllers/api/functions.php:268]: Not sure about the description here. @kodumbeats can we come up with a better phrasing?

**Review by eldadfux — COMMENTED** [review]
Great work, love all the improvments! Left a few minor comments and questions.

File: app/controllers/api/database.php (hunk 1 above)
Comment (TorstenDittmann) [reply app/controllers/api/database.php:152]: In the SDK's - this links need to be absolute 👍🏻

File: app/controllers/api/database.php (hunk 2 above)
Comment (TorstenDittmann) [reply app/controllers/api/database.php]: Adapted these 👍🏻

**Pull request merged**